pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4135,15 +4156,15 @@ private CompletableFuture<Void> createSubscriptions(TopicName topicName, int num
     // Pulsar client-java lib always passes user-agent as X-Java-$version.
     // However, cpp-client older than v1.20 (PR #765) never used to pass it.
     // So, request without user-agent and Pulsar-CPP-vX (X < 1.21) must be rejected
-    private void validateClientVersion() {
+    protected CompletableFuture<Void> internalValidateClientVersionAsync() {
```

Comment:
Question: 
Looks like we don't have the async operation in this method.

Reply:
yes, only make this method to return future.